permissions_validators/src/private_blockchain/mod.rs

```diff
@@ -38,3 +38,541 @@ impl<W: WorldTrait> IsGrantAllowed<W> for ProhibitGrant {
+
+        {
+            let only_accounts_data: IsQueryAllowedBoxed<World> = query::OnlyAccountsData.into();
+
+            assert!(only_accounts_data.check(&alice_id, &op1, &wsv).is_err());
+            assert!(only_accounts_data.check(&alice_id, &op2, &wsv).is_err());
+            assert!(only_accounts_data.check(&carol_id, &op3, &wsv).is_err());
+            assert!(only_accounts_data.check(&alice_id, &op3, &wsv).is_err());
+        }
+    }
+
+    #[test]
+    fn find_assets_by_name() {
+        let TestEnv {
+            alice_id,
+            carol_id,
+            wsv,
+            ..
+        } = TestEnv::new();
+
+        let op1 = QueryBox::FindAssetsByName(FindAssetsByName::new(
+            Name::from_str("gold").expect("Valid"),
+        ));
+        let op2 = QueryBox::FindAssetsByName(FindAssetsByName::new(
+            Name::from_str("silver").expect("Valid"),
+        ));
+        let op3 = QueryBox::FindAssetsByName(FindAssetsByName::new(
+            Name::from_str("bronze").expect("Valid"),
+        ));
+
+        {
+            let only_accounts_domain: IsQueryAllowedBoxed<World> = query::OnlyAccountsDomain.into();
+
+            assert!(only_accounts_domain.check(&alice_id, &op1, &wsv).is_err());
+            assert!(only_accounts_domain.check(&alice_id, &op2, &wsv).is_err());
+            assert!(only_accounts_domain.check(&carol_id, &op3, &wsv).is_err());
+            assert!(only_accounts_domain.check(&alice_id, &op3, &wsv).is_err());
+        }
+
+        {
+            let only_accounts_data: IsQueryAllowedBoxed<World> = query::OnlyAccountsData.into();
+
+            assert!(only_accounts_data.check(&alice_id, &op1, &wsv).is_err());
+            assert!(only_accounts_data.check(&alice_id, &op2, &wsv).is_err());
+            assert!(only_accounts_data.check(&carol_id, &op3, &wsv).is_err());
+            assert!(only_accounts_data.check(&alice_id, &op3, &wsv).is_err());
+        }
+    }
+
+    #[test]
+    fn probihit_register_domains_failed() {
+        let alice_id = AccountId::from_str("alice@test0").expect("Valid");
+
+        let instruction = Instruction::Register(RegisterBox::new(Domain::new(
+            "new_domain".parse().expect("Valid"),
+        )));
+
+        let wsv = WorldStateView::<World>::new(World::new());
+
+        assert!(register::ProhibitRegisterDomains
+            .check(&alice_id, &instruction, &wsv)
+            .is_err());
+    }
+
+    #[test]
+    fn prohibit_register_domains_success() {
+        let alice_id = AccountId::from_str("alice@test0").expect("Valid");
+
+        let instruction = Instruction::Register(RegisterBox::new(Account::new(
+            "bob@test".parse().expect("Valid"),
+            [],
+        )));
+
+        let wsv = WorldStateView::<World>::new(World::new());
+
+        assert!(register::ProhibitRegisterDomains
+            .check(&alice_id, &instruction, &wsv)
+            .is_ok());
+    }
+
+    #[test]
+    fn granted_allowed_register_domains_success() {
```

Comment:
I'd combine the test cases. I'd also give them better names: this one for example is closer to 
`add_register_domains_permission_allows_registering_domain`. You're documenting an invariant, in this case that it A) isn't possible to register the domain without this permission, B) that it becomes possible after adding the permission, C) that it becomes impossible again when you revoke it. 

All three should be checked in a single test.